### ISSUE TITLE
[java] Unused Imports false positive for static import and Standard imports

### ISSUE THREAD

**Issue by vikhil1999** [pmd/pmd#4253]
Affects PMD Version:
6.50.0
Rule: Unused Imports

https://pmd.github.io/pmd-6.4.0/pmd_rules_java_bestpractices.html#unusedimports

Description:
A false positive is generated for the Unused Imports rule on static import and Standard import, even if the methods of that import are being used. Using import static org.mockito.Mockito.*; instead of import static org.mockito.Mockito.mock, pmd shows unused import org.mockito.Mockito. The same issue applies to the standard import as well.

Code Sample demonstrating the issue:

```java
import com.some.thing.models.*; // This is shown as a violation [Incorrect]
import static org.mockito.Mockito.*; // This is shown as a violation [Incorrect]

public class Test {
    private Service service = mock(Service.class);   // mock method comes from mockito import
    private Mapper mapper = mock(Mapper.class);
    XYZ xyz;    // xyz is part of the models import
    ABC abc;    // abc is part of the models import 

    @Before
    public void setup() throws Exception{
        xyz = JsonHelper.convertJsonFileToPojo("MockJson.json", XYZ.class);
        abc= JsonHelper.convertJsonFileToPojo("MockJson.json", ABC.class);
    }
}
```

i have also  added --aux-classPath to the pmd command, as shown below
```
const pmdPath = "/Users/***/pmdPath.txt";
pmd --aux-classpath "${pmdPath}" -d "${this.inputFiles}" -f json -R "${this.path.resolve(this.RulesPath)}" > "${this.pmdReportPath}";
```

pmdPath.txt:
```
/Users/***/.gradle/caches/modules-2/files-2.1/org.mockito/mockito-core/4.9.0/8caeed1bbe5f66a96ff9f8bd28f284709d802930/mockito-core-4.9.0.jar

/Users/***/.gradle/caches/modules-2/files-2.1/org.mockito/mockito-core/3.9.0/3f2009c0b6ec7ed0274491dedd71bc4b9e19ab33/mockito-core-3.9.0-sources.jar

/Users/***/.gradle/caches/modules-2/files-2.1/org.mockito/mockito-core/3.9.0/b7573430aea743b26434b44f4f46272af613e660/mockito-core-3.9.0.jar

/Users/***/.gradle/caches/modules-2/files-2.1/org.mockito/mockito-core/3.11.1/2b9ab8ef1dfccdbae1ca6d39a407129c146f833e/mockito-core-3.11.1.jar

/Users/***/.gradle/caches/modules-2/files-2.1/org.mockito/mockito-core/3.11.1/178871b0bd3db59c69fc350cb65575bb84bb5bf4/mockito-core-3.11.1-sources.jar

/Users/***/.gradle/caches/modules-2/files-2.1/org.mockito/mockito-core/4.0.0/7350016b49c3ffca18b5ef4b664e730bd99cd054/mockito-core-4.0.0-sources.jar

/Users/***/.gradle/caches/modules-2/files-2.1/org.mockito/mockito-core/4.0.0/f5195e0c4a45716bbd2d1d29173adbd148acce3a/mockito-core-4.0.0.jar

/Users/***/.gradle/caches/modules-2/files-2.1/org.mockito/mockito-core/4.5.1/c08b438bd19ba98419cf7919ba63e909f75b71c/mockito-core-4.5.1-sources.jar

/Users/***/.gradle/caches/modules-2/files-2.1/org.mockito/mockito-core/4.5.1/ed456e623e5afc6f4cee3ae58144e5c45f3b3bf/mockito-core-4.5.1.jar

/Users/***/.gradle/caches/modules-2/files-2.1/org.mockito/mockito-core/4.7.0/5eeffd46d39ddd0c24666adaebc7ff62776642bd/mockito-core-4.7.0.jar

/Users/***/.gradle/caches/modules-2/files-2.1/org.mockito/mockito-core/4.9.0/8caeed1bbe5f66a96ff9f8bd28f284709d802930/mockito-core-4.9.0.jar
```

Expected outcome:
PMD should not show violation for unused import as the methods of the import is being used in the code.
PMD reports a violation at line 1 and 2 but that's wrong. That's a false positive.

Running PMD through:
[ CLI ]

**Comment by jsotuyod** [comment]
@vikhil1999 it seems this is down to a wrong usage issue:

As per [the docs](https://pmd.sourceforge.io/pmd-6.52.0/pmd_userdocs_cli_reference.html)

> `--aux-classpath <cp>` | Specifies the classpath for libraries used by the source code. This is used to resolve types in source files. The platform specific path delimiter (":" on Linux, ";" on Windows) is used to separate the entries. Alternatively, a single `file:` URL to a text file containing path elements on consecutive lines can be specified.

Since what you are providing **is not** a `file:` URL, it's taking it as the actual classpath. Being a txt file, no classes are found. The actual usage should be

```
const pmdPath = "file:///Users/***/pmdPath.txt";
pmd --aux-classpath "${pmdPath}" -d "${this.inputFiles}" -f json -R "${this.path.resolve(this.RulesPath)}" > "${this.pmdReportPath}";
```

Having said that, your `pmdPath.txt` should not include multiple versions of the library to ensure the right symbols are found. Including sources is unnecessary, but harmless.

**Comment by vikhil1999** [comment]
Thanks for your quick response and valuable information
